rhel/rhcc/parser.go

```diff
@@ -0,0 +1,224 @@
+			v, err := rhctag.Parse(version)
+			if err != nil {
+				continue
+			}
+			// parse severity
+			var severity string
+			if release.Impact != "" {
+				severity = release.Impact
+			} else {
+				severity = vuln.ThreatSeverity
+			}
+			// parse cpe
+			cpe, err := cpe.Unbind(release.Cpe)
+			if err != nil {
+				zlog.Warn(ctx).Msgf("Could not unbind cpe: %s", release.Cpe)
+			}
+			// collect minor keys
+			minorKey, err := v.MinorStart()
+			if err != nil {
+				continue
+			}
+			// initialize and update the minorKey to consolidated release map
+			if versionsByContainer[packageName] == nil {
+				versionsByContainer[packageName] = make(map[rhctag.Version]*consolidatedRelease)
+			}
+			versionsByContainer[packageName][minorKey] = &consolidatedRelease{
+				Cpe:      cpe,
+				Issued:   release.ReleaseDate.time,
+				Severity: severity,
+			}
+			// initialize and update the fixed in versions slice
+			if versionsByContainer[packageName][minorKey].FixedInVersions == nil {
+				vs := make(rhctag.Versions, 0)
+				versionsByContainer[packageName][minorKey].FixedInVersions = &vs
+			}
+			newVersions := versionsByContainer[packageName][minorKey].FixedInVersions.Append(v)
+			versionsByContainer[packageName][minorKey].FixedInVersions = &newVersions
+		}
+
+		// Build the Vulnerability slice
+		for pkg, releasesByMinor := range versionsByContainer {
+			p := &claircore.Package{
+				Name: pkg,
+				Kind: claircore.BINARY,
+			}
+			// sort minor keys
+			minorKeys := make(rhctag.Versions, 0)
+			for k := range releasesByMinor {
+				minorKeys = append(minorKeys, k)
+			}
+			sort.Sort(minorKeys)
+			// iterate minor key map in order
+			for idx, minor := range minorKeys {
+				// sort the fixed in versions
+				sort.Sort(releasesByMinor[minor].FixedInVersions)
+				// The first minor version range should match all previous versions
+				start := minor
+				if idx == 0 {
+					start = rhctag.Version{}
+				}
+				// For containers such as openshift-logging/elasticsearch6-rhel8 we need to match
+				// the first patch release here.
+				firstPatch, _ := releasesByMinor[minor].FixedInVersions.First()
+				r := &claircore.Range{
+					Lower: start.Version(),
+					Upper: firstPatch.Version(),
+				}
+				v := &claircore.Vulnerability{
+					Updater:            u.name,
+					Name:               vuln.Name,
+					Description:        description,
+					Issued:             releasesByMinor[minor].Issued,
+					Severity:           releasesByMinor[minor].Severity,
+					NormalizedSeverity: rhel.NormalizeSeverity(releasesByMinor[minor].Severity),
+					Package:            p,
+					Dist: &claircore.Distribution{
+						DID:  ID,
+						Name: Name,
+						// TODO for containers such as openshift-logging/elasticearch6-rhel8
+						// This matches the CPE of the last entry. However we would probably
+						// need a vulnerability with a unique CPE for each once we start
+						// filtering by CPE from container scanner.
+						CPE: releasesByMinor[minor].Cpe,
```

Comment:
This is in the wrong spot, this is the application's CPE.

Reply:
I took a look at the security data, and I found we only publish multiple container image tags with multiple CPEs in 0.5% of cases. Therefore I removed CPE from here.

As I mentioned [ealier](https://github.com/quay/claircore/pull/518#discussion_r783625460), I wasn't sure where I should store the CPE to be used in a matcher, so I put it in distribution. If we need to start using CPE in future, we'll have to figure out where to store it then.